### ISSUE TITLE
Future Request: Bookmark/Contents sidebar auto hide

### ISSUE THREAD

**Issue by nobk** [sumatrapdfreader/sumatrapdf#552]
F12 hotkey can hide bookmark sidebar to save screen width for pages, but not handy to mouse using users, F12 also need right hand pressing which conflict with mouse.
I think add an auto hide option can make book pages jump easier to mouse users. And keyboard user need also using mouse click to select chapter after F12 open the bookmark sidebar.

**Comment by SumatraPeter** [comment]
Not that I'm saying this is a bad idea, but since you say that hiding the bookmark sidebar is "not handy to mouse using users", personally I don't find it all that difficult to just move the cursor and click the 'X' button on the sidebar.

**Comment by nobk** [comment]
non-autohide sidebar is not flat side bar, which using visible area, reduce page width, and always occupy visible area until you click the 'X' button.
Click 'X' can close sidebar, but there is no short way open it by using mouse.

**Comment by SumatraPeter** [comment]
>  Click 'X' can close sidebar, but there is no short way open it by using mouse.
That's true. All such issues can IMO be taken care of if the program had a user-customizable toolbar to which any menu command could be added, but that won't happen unless some dev contributes the required code here or adds the feature to his own fork.

**Comment by GitHubRulesOK** [comment]
@kjk secondary suggestion to mouse click show / hide favourites has recently been introduced

The primary suggestion to "auto-hide" is discussed in other issues thus this suggestion could be consolidated with say pop-up / overlay / floating suggestions in #1406

**Comment by A205B064** [comment]
**Proposal: a `SidebarPushContentMode` setting so the sidebar doesn't steal page width**

This thread's core complaint is that the bookmarks/contents sidebar permanently reduces page width. I've implemented an option for that and wanted to gauge interest before opening a PR.

New advanced setting `SidebarPushContentMode` with three values:

- `always` — current behavior; opening the sidebar pushes the page content aside by the full sidebar width.
- `auto` — the sidebar first overlaps the empty margin beside the page, and only pushes the content if that margin isn't wide enough.
- `never` — the sidebar overlays the page and never pushes content (the *"display Bookmarks overlay above the reading area"* idea from #1406).

Here's a GIF showing the `auto` behavior:

<img width="960" height="540" alt="Image" src="https://github.com/user-attachments/assets/4f946c9b-f75d-4632-8468-bfaa37a17875" />

It's not mouse-triggered auto-hide, but in `auto`/`never` the visible sidebar no longer steals page width, which is the main pain point in this thread. It's also wired to a `CmdChangeSidebarPushMode` command (Command Palette) for switching at runtime.

@kjk would you be open to a PR?

**Comment by GitHubRulesOK** [comment]
@A205B064  not whishing to stop PRs (I am not the developer nor adverse to improvements) but don't think that addresses much of the primary complaints (and there have been many :-)

The comments about mouse users wanting on/off is already addressed by context menu , but is not then addressing auto-hide

<img width="695" height="377" alt="Image" src="https://github.com/user-attachments/assets/b2c90e61-a8f1-41bc-8975-1a0fbed38cf5" />

To NOT affect users' full width view NOR upset eBooks view the consistent requests have been for popout/popover to not NEED a divisor. much like auto toolbar in Windows itself. This is a much greater ask as it is a redesign of how the tree is built and appears and "auto" adjusts thus some ask for  internal overlay and others externalised ! 

<img width="836" height="202" alt="Image" src="https://github.com/user-attachments/assets/9cb04a1b-c914-4d06-a951-ce669ff003b7" />

**Comment by kjk** [comment]
The hard part is deciding how exactly it should work.

I've opened #5676 to handle this in a different way
